From siva@dias.ath.cx Mon Aug 30 17:04:08 2004
Date: Mon, 30 Aug 2004 15:15:30 +0530 (IST)
From: Sivasankar Chander <siva@dias.ath.cx>
Reply-To: schander@mindspring.com
To: sjmudd@pobox.com
Cc: Sivasankar Chander <siva@whitebox.dias.ath.cx>
Subject: [Patch] postfix-dict_db.patch

Hi Simon,
         I'm including a small patch to allow building postfix-1.1.12-5
cleanly with Berkeley db4 and gcc-3.3.3 (gcc-2.96 apparently only
produces a warning). My apologies if this has already been fixed in
later postfix-1.1.x releases, but I couldn't find it in the Changelog.
This is useful for those of us who're sticking with Postfix1 but
moving to more recent distros.


##################### Begin postfix-dict_db.patch ####################
--- postfix-1.1.12/src/util/dict_db.c	Sun Aug 29 19:56:04 2004
+++ postfix-1.1.12-patched/src/util/dict_db.c	Sun Aug 29 19:57:21 2004
@@ -520,7 +520,7 @@
 	msg_fatal("set DB cache size %d: %m", DICT_DB_CACHE_SIZE);
     if (type == DB_HASH && db->set_h_nelem(db, DICT_DB_NELM) != 0)
 	msg_fatal("set DB hash element count %d: %m", DICT_DB_NELM);
-    if ((errno = db->open(db, db_path, 0, type, db_flags, 0644)) != 0)
+    if ((errno = db->open(db, NULL, db_path, 0, type, db_flags, 0644)) != 0) 
 	msg_fatal("open database %s: %m", db_path);
     if ((errno = db->fd(db, &dbfd)) != 0)
 	msg_fatal("get database file descriptor: %m");
###################### End postfix-dict_db.patch #####################

-Sivasankar Chander
NOTE by SJM: The info was received in a comment on the following web page
             http://archives.neohapsis.com/archives/postfix/2002-10/3004.html
